core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceTest.java

```diff
@@ -637,6 +688,89 @@ public void testParam() throws Exception {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+                assertThat(EntityUtils.toString(res.getEntity()), is("GET/JSON"));
+            }
+
+            httpPost = new HttpPost(rule.httpUri("/8/same/path"));
+            httpPost.setHeader(org.apache.http.HttpHeaders.CONTENT_TYPE, "application/json");
+            httpPost.setHeader(org.apache.http.HttpHeaders.ACCEPT, "application/json");
+            try (CloseableHttpResponse res = hc.execute(httpPost)) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+                assertThat(EntityUtils.toString(res.getEntity()), is("POST/JSON/BOTH"));
+            }
+
+            httpGet = new HttpGet(rule.httpUri("/8/same/path"));
+            httpGet.setHeader(org.apache.http.HttpHeaders.ACCEPT, "application/json;q=0.9, text/plain");
+            try (CloseableHttpResponse res = hc.execute(httpGet)) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+                assertThat(EntityUtils.toString(res.getEntity()), is("GET/TEXT"));
+            }
+
+            httpGet = new HttpGet(rule.httpUri("/8/same/path"));
+            httpGet.setHeader(org.apache.http.HttpHeaders.ACCEPT, "application/json;q=0.9, text/plain;q=0.7");
+            try (CloseableHttpResponse res = hc.execute(httpGet)) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+                assertThat(EntityUtils.toString(res.getEntity()), is("GET/JSON"));
+            }
+
+            httpGet = new HttpGet(rule.httpUri("/8/same/path"));
+            httpGet.setHeader(org.apache.http.HttpHeaders.ACCEPT,
+                              "application/json;charset=UTF-8;q=0.9, text/plain;q=0.7");
+            try (CloseableHttpResponse res = hc.execute(httpGet)) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+                assertThat(EntityUtils.toString(res.getEntity()), is("GET/TEXT"));
+            }
+
+            httpGet = new HttpGet(rule.httpUri("/8/same/path"));
+            httpGet.setHeader(org.apache.http.HttpHeaders.ACCEPT,
+                              "application/x-www-form-urlencoded, application/json;charset=UTF-8;q=0.9," +
+                              "text/plain;q=0.7");
+            try (CloseableHttpResponse res = hc.execute(httpGet)) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
+                assertThat(EntityUtils.toString(res.getEntity()), is("GET/TEXT"));
+            }
+
+            // No matched media type. It would be handled by the MyAnnotatedService8#sharedGet.
+            // TODO(hyangtack) It is definitely weird behavior because the request does not contain
+            // 'accept: */*'. But we cannot check it strictly because we didn't have a media type
+            // negotiation feature before. Currently we just assume that the request has a 'accept: */*'.
```

Comment:
Shouldn't we assume `accept: */*` only when no `accept` header is present in a request? I think the correct behavior in this case is to respond with `406 Not Acceptable`. We could also add a test case which sends no `accept` header, which leads to `sharedGet()`.